### ISSUE TITLE
Docstrings for Trial runner and unittest

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2133]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2133|
|Type|enhancement|
|Created|2006-09-27 07:56:21Z|

Already written. Filing the ticket to fulfil all righteousness.

<details><summary>Searchable metadata</summary>

```
trac-id__2133 2133
type__enhancement enhancement
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1159343781000000 1159343781000000
changetime__1159496197000000 1159496197000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml removed owner|
|-|-|

Ready for review in branches:trial-docstrings-2133

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @jml|
|-|-|

No changes in this branch that I can see

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

Yeah. Umm. I did something that broke Combinator.

I originally wrote the branch in trial-docstrings-1604. Then I realised that [#11159](https://github.com/twisted/twisted/issues/11159) had been closed, so I svn moved the branch to trial-docstrings-2133. Combinator, thanks to the magic of --stop-on-copy, therefore believes that no changes have been made.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Merged forward to trial-docstrings-2133-2.

Excellent changes.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [18318]) Document a great deal of trial.

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#2133](https://github.com/twisted/twisted/issues/2133)

Docstrings for the package, runner and unittest.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
